### Ans - 1

#### 1. Mean
#### 2. Median
#### 3. Mode

### Ans - 2

#### The difference between mean, median and mode are: Mean is the average value of the given observations. Median is the middle value of the given observations. Mode is the most repeated value in the given observation.
#### Mean tells about average of the data but it is not feasible for outliers
#### Median tells about middle value and gives actucal central value even if we are having outliers.
#### Mode tells about where the most data or frequency lies in the data.

### Ans - 3

In [12]:
import numpy as np
from scipy.stats import mode

arr = np.array([178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5])
arr_mean = np.mean(arr)
arr_median = np.median(arr)
arr_mode = mode(arr)

print(arr_mean, arr_median, arr_mode)

177.01875 177.0 ModeResult(mode=array([177.]), count=array([3]))


### Ans - 4

In [13]:
arr2 = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
arr2_std = np.std(arr2)
arr2_std

1.7885814036548633

### Ans - 5

#### Standard deviation is a statistic that measures the dispersion of a dataset relative to its mean and is calculated as the square root of the variance. The standard deviation is calculated as the square root of variance by determining each data point's deviation relative to the mean.

In [14]:
data = [1,23,1,21,23,3,45,3,32,44,54,322]
print(np.mean(data), np.median(data), mode(data))

47.666666666666664 23.0 ModeResult(mode=array([1]), count=array([2]))


### Ans - 6

#### A Venn diagram is an illustration that uses circles to show the relationships among things or finite groups of things.

### Ans - 7

In [15]:
d1 = {2,3,4,5,6,7}
d2 = {0,2,6,8,10}

inn = d1.intersection(d2)
un = d1.union(d2)

print(inn, un)

{2, 6} {0, 2, 3, 4, 5, 6, 7, 8, 10}


### Ans - 8

#### Skewness is a measurement of the distortion of symmetrical distribution or asymmetry in a data set. Skewness is demonstrated on a bell curve when data points are not distributed symmetrically to the left and right sides of the median on a bell curve.

### Ans - 9

#### If a data is right skewed then median is always greater than mean.

### Ans - 10

#### Both covariance and correlation measure the relationship and the dependency between two variables.
#### Covariance denotes dependency in between -infinite to +infinite. and Correlation denotes dependency in between -1 to +1. 
#### Covariance indicates the direction of the linear relationship between variables.
#### Correlation measures both the strength and direction of the linear relationship between two variables.
#### Correlation values are standardized.
#### Covariance values are not standardized.

#### How used :
#### It helps to find the relationship between two variables.
#### Like if the year increase the age will also going to be increase.
#### So we came to conclusion that year is strongly dependent on the age and vice versa.

### Ans - 11

#### Formula of sample mean = ( Sum of all the numbers / count of the number )

In [16]:
import numpy as np
data = [12,23,4,2,33,22,1,3,4,6,5]
np.array(data).mean()

10.454545454545455

### Ans - 12

#### For normal distributed data the measure of central tendency mean = median = mode. 

### Ans - 13

#### Covariance shows you how the two variables differ, whereas correlation shows you how the two variables are related.

### Ans - 14

#### Outlier mostly affect the mean then little bit median and mode.
#### In measure of dispersion it mostly affect the range then quartile deviation is not affected by the extreme values of the series.

In [17]:
values =  np.array([12, 23,  4,  2, 33, 22,  1,  3,  4,  6,  5, 100])
# here 100 is an outlier so by finding our mean would contains the outliers.
val_mean = values.mean()
val_mean

17.916666666666668

In [19]:
val_mode = mode(values)
val_mode

ModeResult(mode=array([4]), count=array([2]))

In [21]:
val_median = np.median(values)
val_median

5.5

#### Here we can see that our median is 5.5 and our mean is 17.9 means if we draw an histogram based on mean then it will be right skewed because of outlier. and if we draw histogram based on median then possibility we will get normaly distributed graph.

In [23]:
values.var()

708.4097222222222

In [24]:
np.cov(values)

array(772.81060606)